api-spec/protobuf/ark/v1/indexer.proto

```diff
@@ -0,0 +1,222 @@
+    };
+  };
+  rpc GetSpendableVtxos(GetSpendableVtxosRequest) returns (GetSpendableVtxosResponse) {
+    option (google.api.http) = {
+      get: "/v1/spendableVtxos/{address}"
+    };
+  };
+  rpc GetTransactionHistory(GetTransactionHistoryRequest) returns (GetTransactionHistoryResponse) {
+    option (google.api.http) = {
+      get: "/v1/history/{address}"
+    };
+  };
+  rpc GetVtxoChain(GetVtxoChainRequest) returns (GetVtxoChainResponse) {
+    option (google.api.http) = {
+      get: "/v1/vtxo/{outpoint.txid}/{outpoint.vout}/chain"
+    };
+  }
+  rpc GetVirtualTxs(GetVirtualTxsRequest) returns (GetVirtualTxsResponse) {
+    option (google.api.http) = {
+      get: "/v1/virtualTx/{txids}"
+    };
+  }
+  rpc GetSweptCommitmentTx(GetSweptCommitmentTxRequest) returns (GetSweptCommitmentTxResponse) {
+    option (google.api.http) = {
+      get: "/v1/commitmentTx/{txid}/swept"
+    };
+  }
+}
+
+message GetCommitmentTxRequest {
+  string txid = 1;
+}
+message GetCommitmentTxResponse {
+  int64 started_at = 1;
+  int64 ended_at = 2;
+  map<uint32, IndexerBatch> batches = 3;
+}
+
+message GetVtxoTreeRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVtxoTreeResponse {
+  repeated IndexerNode vtxo_tree = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetForfeitTxsRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetForfeitTxsResponse {
+  repeated string txs = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetConnectorsRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetConnectorsResponse {
+  repeated IndexerNode connectors = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetSpendableVtxosRequest {
+  string address = 1;
+  IndexerPageRequest page = 2;
+}
+message GetSpendableVtxosResponse {
+  repeated IndexerVtxo vtxos = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetTransactionHistoryRequest {
+  string address = 1;
+  int64 start_time = 2;
+  int64 end_time = 3;
+  IndexerPageRequest page = 4;
+}
+message GetTransactionHistoryResponse {
+  repeated IndexerTxHistoryRecord history = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetVtxoChainRequest {
+  IndexerOutpoint outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVtxoChainResponse {
+  map<string, IndexerTransactions> graph = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetVirtualTxsRequest {
+  repeated string txids = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVirtualTxsResponse {
+  repeated string txs = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetSweptCommitmentTxRequest {
+  string txid = 1;
+}
+message GetSweptCommitmentTxResponse {
+  repeated string swept_by = 1;
+}
+
+message SubscribeForAddressesRequest {
+  repeated string addresses = 1;
+}
+message SubscribeForAddressesResponse {
+  string address = 1;
+  repeated IndexerVtxo new_vtxos = 2;
```

Comment:
maybe a repeated IndexerOutpoint or even just a txid and let the user fetch through API to keep the streams light?